docs/source/platform-features/message-bus/vip/agent-vip-id.rst

```diff
@@ -0,0 +1,144 @@
+the utility function `vip_main` shall check for the environment variable **AGENT_VIP_IDENTITY**.  If the
+**AGENT_VIP_IDENTITY** environment variable is not set then the `vip_main` function will fall back to a supplied
+identity argument.  `vip_main` will pass the appropriate identity argument to the agent constructor.  If no identity is
+set the Agent class will create a random VIP IDENTITY using python's `uuid4` function.
+
+An agent that inherits from the platform's base Agent class can get it's current VIP IDENTITY by retrieving the value of
+``self.core.identity``.
+
+The primary use of the 'identity' argument to `vip_main` is for agent development.  For development it allows agents to
+specify a default VIP IDENTITY when run outside the platform.  As platform Agents are not started via `vip_main` they
+will simply receive their VIP IDENTITY via the identity argument when they are instantiated.  Using the identity
+argument of the Agent constructor to set the VIP IDENTITY via agent configuration is no longer supported.
+
+At runtime the platform will set the environment variable **AGENT_VIP_IDENTITY** to the value set at installation time.
+
+Agents not based on the platform's base Agent should set their VIP IDENTITY by setting the identity of the ZMQ socket
+before the socket connects to the platform.  If the agent fails to set it's VIP IDENTITY via the ZMQ socket it will be
+selected automatically by the platform.  This platform chosen ID is currently not discoverable to the agent.
+
+
+Agent Implementation
+====================
+
+If an Agent has a preferred VIP IDENTITY (for example the Master Driver Agent prefers to use "platform.driver") it may
+specify this as a default packed value.  This is done by including a file named IDENTITY containing only the desired VIP
+IDENTITY in ASCII plain text in the same directory at the `setup.py` file for the Agent.  This will cause the packaged
+agent wheel to include an instruction to set the VIP IDENTITY at installation time.
+
+This value may be overridden at packaging or installation time.
+
+
+Packaging
+=========
+
+An Agent may have it's VIP IDENTITY configured when it is packaged.  The packaged value may be used by the platform to
+set the **AGENT_VIP_IDENTITY** environment variable for the agent process.
+
+The packaged VIP IDENTITY may be overridden at installation time.  This overrides any preferred VIP IDENTITY of the
+agent.  This will cause the packaged agent wheel to include an instruction to set the VIP IDENTITY at installation time.
+
+To specify the VIP IDENTITY when packaging use the ``--vip-identity`` option when running `volttron-pkg package`.
+
+
+Installation
+============
+
+An agent may have it's VIP IDENTITY configured when it is installed.  This overrides any VIP IDENTITY specified when the
+agent was packaged.
+
+To specify the VIP IDENTITY when packaging use the ``--vip-identity`` option when running `volttron-ctl install`.
```

Comment:
packaging is from vpkg right?  though you use install here?  not sure if this is right or not please check